kernel/src/main/scala/cats/kernel/Semigroup.scala

```diff
@@ -178,6 +178,11 @@ object Semigroup
    */
   @inline def last[A]: Semigroup[A] = (_, y) => y
 
+  @inline def reverse[A](implicit ev: Semigroup[A]): Semigroup[A] = ev.reverse
```

Comment:
I don't see a huge benefit here since to use it you will often need `Semigroup.reverse[String]` compared to `Semigroup[String].reverse` which isn't a huge difference. You could avoid the type by passing to a method that expects a certain type, which may be common (since you are passing to an implicit/using in many cases).

Reply:
Thanks. The reason I had added it was for [consistency with the other methods](https://github.com/typelevel/cats/issues/4615#issuecomment-2159205016).

In any case, I've removed the commit adding `reverse` and kept just the one adding `intercalate`.

Reply:
We could say the rule is: if the method takes a parameter of that is sufficient to infer the type of the result we add this method. In that way, it's still consistent.

Reply:
True. I think that's a pretty good rule. Thanks for your review.